src/permutation_test.js

```diff
@@ -0,0 +1,98 @@
+/* @flow */
+
+import mean from './mean';
+import shuffleInPlace from './shuffle_in_place';
+
+/**
+ * Conducts a [permutation test](https://en.wikipedia.org/wiki/Resampling_(statistics)#Permutation_tests)
+ * to determine if two data sets are *significantly* different from each other, using
+ * the difference of means between the groups as the test statistic. 
+ * The function allows for the following hypotheses:
+ * - two_tail = Null hypothesis: the two distributions are equal.
+ * - greater = Null hypothesis: observations from sampleX tend to be smaller than those from sampleY.
+ * - less = Null hypothesis: observations from sampleX tend to be greater than those from sampleY.
+ * [Learn more about one-tail vs two-tail tests.](https://en.wikipedia.org/wiki/One-_and_two-tailed_tests)
+ *
+ * @param {Array<number>} sampleX first dataset (e.g. treatment data)
+ * @param {Array<number>} sampleY second dataset (e.g. control data)
+ * @param {string} alternative alternative hypothesis, either 'two_sided' (default), 'greater', or 'less'
+ * @param {number} k number of values in permutation distribution. 
+ * @returns {number} p-value The probability of observing the difference between groups (as or more extreme than what we did), assuming the null hypothesis.
+ *
+ * @example
+ * var control = [2, 5, 3, 6, 7, 2, 5];
+ * var treatment = [20, 5, 13, 12, 7, 2, 2];
+ * permutationTest(control, treatment); // ~0.1324 
+ */
+function permutationTest(
+    sampleX/*: Array<number> */,
+    sampleY/*: Array<number> */,
+    alternative/*: string */ = 'two_side',
+    k/*: number */ = 10000)/*: ?number */ {
```

Comment:
In general this looks great - one thing is that this is using a default parameter, which has more [limited browser support](https://kangax.github.io/compat-table/es6/#test-default_function_parameters) than simple-statistics currently has - this would drop IE11 support, for instance. Mind replacing this with an equivalent like

```js
if (k === undefined) {
  k = 10000;
}
```